src/AutoRest.CSharp/Common/Input/InputTypes/Transformation/SubscriptionIdTransformer.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+
+namespace AutoRest.CSharp.Common.Input.InputTypes.Transformation
+{
+    internal class SubscriptionIdTransformer
+    {
+        public static void Transform(InputNamespace inputNamespace)
+        {
+            foreach (var client in inputNamespace.Clients)
+            {
+                foreach (var operation in client.Operations)
+                {
+                    var subscriptionIdParameter = operation.Parameters.FirstOrDefault(p => p.NameInRequest == "subscriptionId");
```

Comment:
we already have a place to deal with subscriptionId, should put the logic there instead
https://github.com/Azure/autorest.csharp/blob/5fac408daac7deb1dd41849fc9150b4c603af747/src/AutoRest.CSharp/Common/Input/InputTypes/Transformation/InputClientTransformer.cs#L35-L47